ScatterPeas/scatterpeas/models.py

```diff
@@ -0,0 +1,632 @@
+
+
+class Reminder(Base):
+    """Reminder table includes payload information for reminders.
+
+    Provides functionality for putting jobs onto the jobs table.
+    Columns include:
+    -id
+    -alias_id: ForeignKey to Alias table
+    -title: Short version of reminder; required
+    -text_payload: To include as email body, optional
+    -media_payload: For including pictures etc. Not yet fully implemented.
+    -rstate: Tracks whether Reminder has pending jobs. True for active.
+    """
+
+    __tablename__ = 'reminders'
+    rrule = relationship("RRule", backref="reminders", uselist=False)
+    jobs = relationship("Job", backref="reminder")
+
+    id = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
+    alias_id = Column(Integer, ForeignKey('aliases.id'), nullable=False)
+    # keeps the rrule currently generating jobs
+    # rrule_id = Column(Integer, ForeignKey('rrules.id'))
+    title = Column(Unicode(256), nullable=False)
+    text_payload = Column(Unicode(20000))
+    # this might end up just being a link to a static resourc; we will see
+    media_payload = Column(Unicode(256))
+    # rstate is used to to track whether a reminder is still producing jobs
+    # turned to False after execution of last job
+    rstate = Column(Boolean)
+
+    @property
+    def __acl__(self):
+        rule_list = []
+        for user in self.alias.users:
+            rule_list.append((Allow, user.username, 'edit'))
+        rule_list.append((Allow, 'group:admin', ALL_PERMISSIONS))
+
+    @classmethod
+    def parse_reminder(cls, reminder_id, session=None):
+        """Takes a reminder and returns the reminder information needed
+        to parse a reminder into a text or email. Also checks for pending jobs,
+        instantiates them. Flips rstate to False if current call is the last"""
+        if session is None:
+            session = DBSession
+        next_job = cls.get_next_job(reminder_id)
+        reminder = cls.retrieve_instance(reminder_id)
+        if next_job is None:
+            reminder.rstate = False
+        else:
+            Job.create_job(reminder_id, next_job)
+        return {"title": reminder.title,
+                "text_payload": reminder.text_payload,
+                "media_payload": reminder.media_payload,
+                "alias_id": reminder.alias_id}
+
+    @classmethod
+    def retrieve_instance(cls, reminder_id, session=None):
+        """Retrieves a reminder instance from reminder_id"""
+        if session is None:
+            session = DBSession
+        return session.query(Reminder).filter(Reminder.id == reminder_id).one()
+
+    @classmethod
+    def get_next_job(cls, reminder_id, session=None):
+        """Takes a reminder_id to retrieve current rrule set and generate
+        the next job in datetime format for utc. Returns None if there is
+        no next job."""
+        if session is None:
+            session = DBSession
+        reminder = cls.retrieve_instance(reminder_id)
```

Comment:
see this call here?  If this were an instance method, you wouldn't need it at all.